src/Parser.php

```diff
@@ -3289,6 +3290,11 @@ private function parseObjectCreationExpression($parentNode) {
             $objectCreationExpression->classMembers = $this->parseClassMembers($objectCreationExpression);
         }
 
+        // PHP8.4 new with no parenthesis
+        if ($this->getCurrentToken()->kind === TokenKind::ArrowToken) {
+            return $this->parsePostfixExpressionRest($objectCreationExpression);
```

Comment:
In line with the rest of this libaray this should probably have a run time `php_version() > 80400` or whatever? 

(incidentally would be good to refactor so that it didn't depend on the runtime PHP version)

Reply:
+1 for being able to configure at what PHP level the parser works